cachix/cachix-deployment/Main.hs

```diff
@@ -30,20 +32,29 @@ import Protolude.Conv
 import System.IO (BufferMode (..), hSetBuffering)
 import qualified Wuss
 
+-- | Activate the new deployment.
+--
+-- If the target profile is already locked by another deployment, exit
+-- immediately and rely on the backend to reschedule.
 main :: IO ()
 main = do
   setLocaleEncoding utf8
   hSetBuffering stdout LineBuffering
   hSetBuffering stderr LineBuffering
+
   input <- escalateAs (FatalError . toS) . Aeson.eitherDecode . toS =<< getContents
-  CachixWebsocket.runForever (CachixWebsocket.websocketOptions input) (handleMessage input)
+  let profile = toS . CachixWebsocket.profile . Input.websocketOptions $ input
+  void $
+    Lock.withTryLock ("/nix/var/nix/profiles/" <> profile) $
```

Comment:
We should have the agent figure out the type of deployment (nixos, nix-darwin, etc) and provide this info to the deployer.